src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -90,17 +100,21 @@ function SyntheticEvent(dispatchConfig, targetInst, nativeEvent, nativeEventTarg
 assign(SyntheticEvent.prototype, {
 
   preventDefault: function() {
+    if (this._destructored) {
+      if (__DEV__) {
+        warning(
+          event,
+          'This synthetic event is reused for performance reasons. If you\'re ' +
+          'seeing this, you\'re calling `preventDefault` on a ' +
+          'released/nullified synthetic event. This is a no-op. See ' +
+          'https://fb.me/react-event-pooling for more information.'
+        );
+      }
+      return;
+    }
+
     this.defaultPrevented = true;
     var event = this.nativeEvent;
-    if (__DEV__) {
```

Comment:
The thing I'm trying to avoid is multiple warnings. If we try to set `this.defaultPrevented` after this has been destructored then we'll get a warning for that action. Similarly for accessing `this.nativeEvent`. So this code change was an effort to first check if it's already been destructored, if it has, don't attempt to access/set variables, log the one warning, and return.